daemon/algod/api/server/v2/dryrun_test.go

```diff
@@ -393,6 +393,25 @@ func checkAppCallPass(t *testing.T, response *generated.DryrunResponse) {
 	}
 }
 
+func checkAppCallScratchType(t *testing.T, response *generated.DryrunResponse, slot int, tt basics.TealType) {
```

Comment:
I'd rather see this check function take the txn index, that way it can assert that txn in question has an AppCallTrace. And then it should assert that traceline.Scratch is not nil, and that the slot has the right type.  As written, it's too lenient.  It allows allows the scratch to not be set, or all of the txns to be of the wrong type, etc.

Reply:
It might also be useful to have a way to assert that a scratch slot is not included in the response. If I understand correctly, though point of all this `maxActive` stuff is to keep the responses small, and not contain slots that were never touched, right?  We should test that is done right.

Reply:
Correct, I'll make both changes

Reply:
Perhaps you're dealing with the fact that at the start of the trace, not type is set, since it hasn't been used yet.  In that case, I'd still take the txn index, and confirm AppCallTrace, but I'd leave the other `continue`s. But, want to be sure that line 408 happens at least once, or we're not really checking anything.  So I'd set a flag there, and assert it at the end of the function.